src/ses.c

```diff
@@ -462,12 +463,37 @@ int ses_send_diag(int fd, struct ses_pages *sp)
 			       sp->page2.len, 0, debug);
 }
 
+static void get_led_status(struct ses_pages *sp, int idx, enum ibpi_pattern *led_status)
+{
+	struct ses_slot_ctrl_elem *descriptors = (void *)(sp->page2.buf + 8);
+	struct ses_slot_ctrl_elem *desc_element = NULL;
+	descriptors++;
+	desc_element = &descriptors[idx];
+
+	*led_status = IBPI_PATTERN_NORMAL;
+
+	if (desc_element->b2 & 0x02) {
+		*led_status = IBPI_PATTERN_LOCATE;
+	}
+	if (desc_element->b3 & 0x60) {
+		*led_status = IBPI_PATTERN_FAILED_DRIVE;
+	}
+}
+
 int ses_get_slots(struct ses_pages *sp, struct ses_slot **out_slots, int *out_slots_count)
 {
 	unsigned char *add_desc = NULL;
 	unsigned char *ap = NULL, *addr_p = NULL;
 	int i, j, len = 0;
 
+	// Our expectation is that the caller is passing us NULL or a valid chunk of memory for
+	// slots that was used before and are now reloading, so free it first.
+	if (*out_slots) {
+		free(*out_slots);
```

Comment:
please move freeing below,

just above 
```c
*out_slots = slots;
```

also, checking 
`if (*out_slots) {`
is unnecessary as free() will do nothing if  `*out_slots` is NULL

Reply:
Done